### ISSUE TITLE
Replace Paralysis() on guns with a fire_act_delay() instead

### ISSUE THREAD

**Issue by JohnBigless** [BadDeathclaw/TG-Claw#702]
When a ranged weapon (things like guns) are picked up, the player will get stunned.
The reason on why this is a thing has been told by Cassius.
![image](https://user-images.githubusercontent.com/44721288/54439656-874a1d00-4739-11e9-9738-f341e6d77f41.png)
![image](https://user-images.githubusercontent.com/44721288/54439677-90d38500-4739-11e9-83e3-e4a15f8cbbe8.png)
Since there is no gameplay reason on why this is a thing, it should probably be removed and some sort of admin log spam prevention system should be put in place.
![image](https://user-images.githubusercontent.com/44721288/54439967-1d7e4300-473a-11e9-9132-c75bda8aff82.png)

**Comment by RenGusta** [comment]
You're aware it's a balancing point for escalation, on that it prevents quickdraw assblasts when you tell someone that they're being robbed.

**Comment by GhostEcho** [comment]
Yeah Hyena only explained one aspect of it and not the actual gameplay aspect, both are valid enough though Im not opposed to making pistol holsters actually useful for something like quickdrawing

**Comment by JohnOxford** [comment]
>make shitty work around
>you can completely ignore the work around by just carrying the gun in your hand before you rob someone or always in the case that you are the person getting robbed
>shitty work around no longer works

valid options:
add a aiming system, similar to bay, you don't have to kill someone if they know they're going to die because your aiming at them and they comply with your demands, but if they think they have a chance to get away, odds are they'll move and not comply

remove this all together and just understand the fact that robbing people is dangerous and their ability to pull out a gun and shoot you is easily achievable, more so in a time and place where everyone in the game has a gun at round start

something something hugbox

**Comment by SpaceVampire** [comment]
The aiming system idea sounds pretty cool, lets try that. Like you said a big problem with the current system is you can just have your gun already out.

**Comment by UndeadNappist** [comment]
It doesn't spam the admins with logs, it adds a log to your attack log when you draw and puts a message in the chat.

Paralysis() isn't stun - stun makes you drop all items you're holding and, I believe, puts you prone (along with a few other side effects).

The _reason_ I added Paralysis is separate from the combat log thing - the latter was explicitly to make admins' jobs easier without being annoying (again, it goes to attack logs, NOT spamming all of the admins with a visible message) was because there was a huge issue with people quickdrawing weapons and 100-0ing someone who was in the middle of talking without any chance for them to fight back.

This exasperated the current problems with escalation rules further, and, frankly, it doesn't feel 'fair' from a gameplay perspective to have a conversation and then just be on the floor in the blink of an eye. Lag can make this worse.

I had at the time three options:
1. Add a gundraw effect (the one I invariably chose)
2. Go through and nerf every gun in the game (which would actually punish Robusto, The Lone Gunman, because suddenly he couldn't use his weapon's damage as effectively as one would like AGAINST ALL ODDS)
3. Double or triple player health (which I determined wouldn't actually be an effective solution and cause other issues besides - along with what I suspect numerous bugs due to spaghetti code).

The aiming mechanic is a cool idea but the implementation of it is not good at this time - besides which, taking damage in Bay is entirely different than in TG, and people move a LOT slower in Bay... but here, there is no RNG involved with actually landing shots, which I greatly approve of.

**Comment by UndeadNappist** [comment]
tl;dr the basis of your argument is wrong, stop being rude

cross-ref for admin spam prevention https://github.com/BadDeathclaw/TG-Claw/issues/622

Into actual feedback, it feels like the biggest problem with the current system is it punishes people who like to use pistols too hard, and that it shouldn't use Paralysis() at all. It should use a step or act delay that prevents you from _firing_ after drawing, not leave you helpless. It's annoying when trying to do something simple like inventory management but, at this time, I'm not quite sure how to do that.

**Comment by UndeadNappist** [comment]
Done.
